osu.Game/Utils/ModUtils.cs

```diff
@@ -112,7 +112,16 @@ public static bool CheckAllowed(IEnumerable<Mod> combination, IEnumerable<Type>
         /// <param name="invalidMods">Invalid mods, if any were found. Will be null if all mods were valid.</param>
         /// <returns>Whether the input mods were all valid. If false, <paramref name="invalidMods"/> will contain all invalid entries.</returns>
         public static bool CheckValidForGameplay(IEnumerable<Mod> mods, [NotNullWhen(false)] out List<Mod>? invalidMods)
-            => checkValid(mods, m => m.Type != ModType.System && m.HasImplementation && !(m is MultiMod), out invalidMods, true);
+        {
+            mods = mods.ToArray();
+
+            // exclude multi mods from compatibility checks.
+            // the loop below automatically marks all multi mods as not valid for gameplay anyway.
```

Comment:
```suggestion
            // the `checkValid()` call below marks all multi mods as not valid for gameplay anyway.
```

But this may potentially change, depending on what happens with the other review conversation (potentially it will be "the `checkValid()` call above").

Reply:
As per the suggestion in the other review comment, I've moved this comment to the `checkValid` method and briefly explained why it's separated from the post-compatibility one.